File: Boletines/General/pruebascpp/src/ServicioServidor.cpp
#include <ros/ros.h>
#include <pruebascpp/MensajeServicio.h>

#include <iostream>
using namespace std;

/*
bool callback(pruebascpp::MensajeServicioRequest &req, pruebascpp::MensajeServicioResponse &res){
    cout << "Peticion: " << req.Peticion << endl;
    res.Resultado = req.Peticion * 2;
    return true;
}


int main (int argc, char ** argv){

    // Creacion del nodo principal
    ros::init(argc, argv, "servidor_servicio");
    ros::NodeHandle nh;

    // Creacion del servidor
    ros::ServiceServer servidor = nh.advertiseService<pruebascpp::MensajeServicioRequest, pruebascpp::MensajeServicioResponse>("Servicio", callback);
    ROS_INFO("Servidor creado");

    ros::MultiThreadedSpinner spinner(4);
    spinner.spin();
    return 0;
}
*/

class Servidor{
    private:
        ros::NodeHandle nh;
        ros::ServiceServer servidor;

        bool callback (pruebascpp::MensajeServicioRequest &req, pruebascpp::MensajeServicioResponse &res){
            cout << "Peticion: " << req.Peticion << endl;
            res.Resultado = req.Peticion * 2;
            return true;
        }
    
    public:
        Servidor(){
            servidor = nh.advertiseService("Servicio", &Servidor::callback, this);
        }
};


int main(int argc, char ** argv ){
    // Creacion del nodo principal
    ros::init(argc, argv, "servidor_servicio");
    ros::NodeHandle nh;
    
    //Creacion del objeto de servicio
    Servidor servidor;
    ros::spin();
    return 0;
}

File: Boletines/General/pruebas2/src/ServicioServidor.py
#!/usr/bin/env python3
import rospy 
from pruebas2.srv import MensajeServicio, MensajeServicioResponse



def callback(request):
    print("Peticion: ", request.Peticion)
    respuesta = MensajeServicioResponse()
    respuesta.Resultado = request.Peticion * 2
    return respuesta

#Creacion del nodo principal
rospy.init_node('servicio')
rospy.loginfo("Nodo servicio activado")

#Creacion del servicio
mi_servicio = rospy.Service('Servicio', MensajeServicio, callback)
rospy.loginfo("Servicio activado")

rospy.spin()